clippy_lints/src/suspicious_xor.rs

```diff
@@ -0,0 +1,55 @@
+use clippy_utils::{numeric_literal::NumericLiteral, source::snippet};
+use rustc_errors::Applicability;
+use rustc_hir::{BinOpKind, Expr, ExprKind, Lit};
+use rustc_lint::{LateContext, LateLintPass, LintContext};
+use rustc_middle::lint::in_external_macro;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Warns for a Bitwise XOR operator being probably confused as a powering. It will not trigger if any of the numbers are not in decimal.
+    /// ### Why is this bad?
+    ///	It's most probably a typo and may lead to unexpected behaviours.
+    /// ### Example
+    /// ```rust
+    /// let x = 3_i32 ^ 4_i32;
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// let x = 3_i32.pow(4);
+    /// ```
+    #[clippy::version = "1.65.0"]
+    pub SUSPICIOUS_XOR,
```

Comment:
IMO it would be better to point out misuse as a power, so its lint name should include `pow`. For example, `suspicious_xor_as_pow` or `xor_used_as_pow` from https://github.com/rust-lang/rust-clippy/pull/6182.